Hexagonal division object

GiudGiud opened the pull request:
refs #26225

Moving some code from Cardinal, notably a bunch of utils. 
I think (and got consensus from a lot of app devs doing hex reactor modeling) it makes sense to have them in MOOSE so they can be relied upon by the apps. That way the conventions for numbering pins and channels are shared, and we dont redo the same work over and over

GiudGiud committed:
Add unit tests to the reactor module to prepare for hexlattice tests, refs #26225

aprilnovak committed:
Move unit test for Cardinal hex lattice utility
refs #26225

aprilnovak committed:
Move hexagonal lattice utility from Cardinal with a slight rename to avoid conflict

aprilnovak committed:
Add geometry utilities to support Hex utilites

aprilnovak committed:
Move unit tests for geometry uilities, refs #26225

GiudGiud committed:
Add hex grid division

GiudGiud committed:
Add more functions to util for dealing with hex lattices

GiudGiud committed:
Add test for hex grid division

Add an option to make the mesh finer in the test to check results visually

GiudGiud committed:
Work on the Z axis projection. This is still not quite there

GiudGiud committed:
Improve docstrings on hex lattice util, make a few member attributes clearer

GiudGiud committed:
Find closest pin properly when inside the hex lattice
Rework all for loops for C++11

GiudGiud committed:
Remove Cardinal copyright and use MOOSE copyright>
This is allowed because both LGPL and also I have permission from April

GiudGiud committed:
Add documentation for the hexgrid division
Add more detials to other grids

GiudGiud committed:
Try to silence compile warnings

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1940890/) on e3f207f wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26311/site)

This comment will be updated on new commits.

GiudGiud committed:
Fix num divisions for mesh divisions distributed over positions

GiudGiud committed:
Check mesh division overlap within tolerances
Register to correct app
Use z_index consistently

GiudGiud commented on framework/include/utils/GeometryUtils.h:
```diff
@@ -0,0 +1,156 @@
+ */
+Real minDistanceToPoints(const Point & pt,
+                         const std::vector<Point> & candidates,
+                         const unsigned int & axis);
```
I think we will want an enum here instead of unsigned int

GiudGiud replied:
it's actually kind of nice to be able to acess points directly at the axis. Probably better for performance too. So I m happy with this unsigned int

lindsayad replied:
```suggestion
                         const unsigned int axis);
```

GiudGiud committed:
Avoid risky reference for members of lattice util
Dont try a recover on a steady solve

GiudGiud committed:
Admit defeat with the compile warning treated as error
cast all comparisons to the same type

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1967320/) on de14809 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>4d71af</th>
      <th colspan=4>#26311 de1480</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.07%</td>
      <td>85.07%</td>
      <td>+0.00%</td>
      <td>88.12%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>98385</td>
      <td>98520</td>
      <td>+135</td>
      <td>141</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>17265</td>
      <td>17284</td>
      <td>+19</td>
      <td>19</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26311/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26311/coverage/framework)


## Modules coverage

### Reactor

<table>
  <thead>
    <tr>
      <th></th>
      <th>4d71af</th>
      <th colspan=4>#26311 de1480</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>95.22%</td>
      <td>94.30%</td>
      <td>-0.92%</td>
      <td>84.21%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>5137</td>
      <td>5589</td>
      <td>+452</td>
      <td>448</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>258</td>
      <td>338</td>
      <td>+80</td>
      <td>84</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26311/coverage_diff/reactor)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26311/coverage/reactor)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26311/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 88.12% is less than the suggested 90.0%
- `reactor` new line coverage rate 84.21% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud commented:
Failures in sweep coming from THM and unrelated

lindsayad commented on framework/include/utils/GeometryUtils.h:
```diff
@@ -0,0 +1,156 @@
+ * @param[in] axis axis perpendicular to the projection plane
+ * @return indices of coordinates on plane
+ */
+std::pair<unsigned int, unsigned int> projectedIndices(const unsigned int & axis);
```
principle should be applied throughout this file and its associated `.C`

lindsayad commented on framework/include/utils/GeometryUtils.h:
```diff
@@ -0,0 +1,156 @@
+ * @param[in] axis axis index (0 = x, 1 = y, 2 = z) perpendicular to the projection plane
+ * @return distance from line
+ */
+Real projectedDistanceFromLine(Point pt, Point line0, Point line1, const unsigned int & axis);
```
I see looking in the .C that there is actual modification of some (all?) of pass-by-value `Points`, so ignore my comment where appropriate (possibly everywhere)

lindsayad commented on framework/include/utils/GeometryUtils.h:
```diff
@@ -0,0 +1,156 @@
+ * @param[in] axis axis perpendicular to plane onto which point and line are first projected
+ * @return half space of line
+ */
+Real projectedLineHalfSpace(Point pt1, Point pt2, Point pt3, const unsigned int & axis);
```
Ah these points get modified, nvm

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+  std::vector<bool> positive_half_space;
+  for (const auto i : index_range(corners))
+  {
+    int next = (i == n_pts - 1) ? 0 : i + 1;
+    auto half = projectedLineHalfSpace(point, corners[i], corners[next], axis);
```
```suggestion
    const int next = (i == n_pts - 1) ? 0 : i + 1;
    const auto half = projectedLineHalfSpace(point, corners[i], corners[next], axis);
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+bool
+pointOnEdge(const Point & point, const std::vector<Point> & corners, const unsigned int & axis)
+{
+  auto n_pts = corners.size();
+  auto idx = projectedIndices(axis);
+
+  Real tol = 1e-8;
```
```suggestion
  const auto n_pts = corners.size();
  const auto idx = projectedIndices(axis);

  constexpr Real tol = 1e-8;
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+
+    // check that the point is "between" the two points; TODO: first pass
+    // we can just compare x and y coordinates
+    bool between_points = (point(idx.first) >= std::min(pt1(idx.first), pt2(idx.first))) &&
```
```suggestion
    const bool between_points = (point(idx.first) >= std::min(pt1(idx.first), pt2(idx.first))) &&
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+  pt1(axis) = 0.0;
+  pt2(axis) = 0.0;
+
+  auto i = projectedIndices(axis);
+
+  Real dx = pt2(i.first) - pt1(i.first);
+  Real dy = pt2(i.second) - pt1(i.second);
+
+  Point normal = projectPoint(dy, -dx, axis);
+  Point gap_line = pt2 - pt1;
+
+  auto cross_product = gap_line.cross(normal);
```
```suggestion
  const auto i = projectedIndices(axis);

  const Real dx = pt2(i.first) - pt1(i.first);
  const Real dy = pt2(i.second) - pt1(i.second);

  const Point normal = projectPoint(dy, -dx, axis);
  const Point gap_line = pt2 - pt1;

  const auto cross_product = gap_line.cross(normal);
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+  if (cross_product(axis) > 0)
+    return normal.unit();
+  else
+  {
+    Point corrected_normal = projectPoint(-dy, dx, axis);
+    return corrected_normal.unit();
+  }
```
```suggestion
    return projectPoint(-dy, dx, axis).unit();
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+polygonCorners(const unsigned int & num_sides, const Real & radius, const unsigned int & axis)
+{
+  std::vector<Point> corners;
+  Real theta = 2.0 * M_PI / num_sides;
+  Real first_angle = M_PI / 2.0 - theta / 2.0;
```
```suggestion
  const Real theta = 2.0 * M_PI / num_sides;
  const Real first_angle = M_PI / 2.0 - theta / 2.0;
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+
+  for (const auto i : make_range(num_sides))
+  {
+    Real angle = first_angle + i * theta;
+    Real x = radius * cos(angle);
+    Real y = radius * sin(angle);
```
```suggestion
    const Real angle = first_angle + i * theta;
    const Real x = radius * cos(angle);
    const Real y = radius * sin(angle);
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+Point
+rotatePointAboutAxis(const Point & p, const Real & angle, const Point & axis)
+{
+  Real cos_theta = cos(angle);
+  Real sin_theta = sin(angle);
```
```suggestion
  const Real cos_theta = cos(angle);
  const Real sin_theta = sin(angle);
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+  Real sin_theta = sin(angle);
+
+  Point pt;
+  Real xy = axis(0) * axis(1);
+  Real xz = axis(0) * axis(2);
+  Real yz = axis(1) * axis(2);
```
```suggestion
  const Real xy = axis(0) * axis(1);
  const Real xz = axis(0) * axis(2);
  const Real yz = axis(1) * axis(2);
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+  Real xz = axis(0) * axis(2);
+  Real yz = axis(1) * axis(2);
+
+  Point x_op(cos_theta + axis(0) * axis(0) * (1.0 - cos_theta),
+             xy * (1.0 - cos_theta) - axis(2) * sin_theta,
+             xz * (1.0 - cos_theta) + axis(1) * sin_theta);
+
+  Point y_op(xy * (1.0 - cos_theta) + axis(2) * sin_theta,
+             cos_theta + axis(1) * axis(1) * (1.0 - cos_theta),
+             yz * (1.0 - cos_theta) - axis(0) * sin_theta);
+
+  Point z_op(xz * (1.0 - cos_theta) - axis(1) * sin_theta,
```
```suggestion
  const Point x_op(cos_theta + axis(0) * axis(0) * (1.0 - cos_theta),
             xy * (1.0 - cos_theta) - axis(2) * sin_theta,
             xz * (1.0 - cos_theta) + axis(1) * sin_theta);

  const Point y_op(xy * (1.0 - cos_theta) + axis(2) * sin_theta,
             cos_theta + axis(1) * axis(1) * (1.0 - cos_theta),
             yz * (1.0 - cos_theta) - axis(0) * sin_theta);

  const Point z_op(xz * (1.0 - cos_theta) - axis(1) * sin_theta,
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+boxCorners(const BoundingBox & box, const Real & factor)
+{
+  Point diff = (box.max() - box.min()) / 2.0;
+  Point origin = box.min() + diff;
```
```suggestion
  const Point origin = box.min() + diff;
```

lindsayad commented on framework/src/utils/GeometryUtils.C:
```diff
@@ -0,0 +1,283 @@
+  diff *= factor;
+
+  // Vectors for sides of box
+  Point dx(2.0 * diff(0), 0.0, 0.0);
+  Point dy(0.0, 2.0 * diff(1), 0.0);
+  Point dz(0.0, 0.0, 2.0 * diff(2));
```
```suggestion
  const Point dx(2.0 * diff(0), 0.0, 0.0);
  const Point dy(0.0, 2.0 * diff(1), 0.0);
  const Point dz(0.0, 0.0, 2.0 * diff(2));
```

lindsayad commented on modules/reactor/doc/content/source/meshdivisions/HexagonalGridDivision.md:
```diff
@@ -0,0 +1,29 @@
+
+Using a [Positions](syntax/Positions/index.md) object as the [!param](/MeshDivisions/HexagonalGridDivision/center_positions)
+parameter, multiple hexagonal grids can be created around each position computed by that object. The division index
+returned is for a point closest to position $i$ is then:
```
```suggestion
returned for a point closest to position $i$ is then:
```

lindsayad commented on modules/reactor/include/utils/HexagonalLatticeUtils.h:
```diff
@@ -0,0 +1,823 @@
+#include "Moose.h"
+#include "MooseTypes.h"
+
+#ifndef CARDINAL_ENUM_H
+namespace channel_type
```
If this header gets included before `CardinalEnum.h`, then there will still be trouble

GiudGiud replied:
umm not much I can do about it though. Next time we update moose in cardinal we ll just have to consolidate the utils

lindsayad commented on modules/reactor/src/meshdivisions/HexagonalGridDivision.C:
```diff
@@ -0,0 +1,194 @@
+  // Most of the infrastructure already exists to use another axis, it's just not disabled because
+  // not tested.
+  MooseEnum z_axis("X Y Z", "Z");
+  params.addParam<MooseEnum>("z_axis", z_axis, "Axial direction for the lattice");
+  params.suppressParameter<MooseEnum>("z_axis");
```
I don't get this. why have this parameter?

GiudGiud replied:
I can remove it

GiudGiud replied:
but I started out wanting to support any axis. Then realized the utility is hardcoded to error on X and Y because it's not tested so surely not working

lindsayad commented on modules/reactor/src/meshdivisions/HexagonalGridDivision.C:
```diff
@@ -0,0 +1,194 @@
+  // Check that the grid is well-defined
+  if (_center_positions)
+  {
+    Real min_center_dist = _center_positions->getMinDistanceBetweenPositions();
```
```suggestion
    const Real min_center_dist = _center_positions->getMinDistanceBetweenPositions();
```

lindsayad commented on modules/reactor/src/meshdivisions/HexagonalGridDivision.C:
```diff
@@ -0,0 +1,194 @@
+          min_center_dist,
+          "), closer than the extent of each grid (",
+          _lattice_flat_to_flat,
+          "). Mesh division is ill - defined ");
```
```suggestion
          "). Mesh division is ill-defined ");
```

lindsayad commented on modules/reactor/src/meshdivisions/HexagonalGridDivision.C:
```diff
@@ -0,0 +1,194 @@
+    ir = _hex_latt->closestPinIndex(pc);
+
+  // Find axial index
+  const auto not_found = MooseMeshDivision::INVALID_DIVISION_INDEX;
```
```suggestion
  constexpr auto not_found = MooseMeshDivision::INVALID_DIVISION_INDEX;
```

GiudGiud replied:
not compiling
```
/Users/c/projects/moose_v1/modules/reactor/src/meshdivisions/HexagonalGridDivision.C:158:18: error: constexpr variable 'not_found' must be initialized by a constant expression
  constexpr auto not_found = MooseMeshDivision::INVALID_DIVISION_INDEX;
                 ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/c/projects/moose_v1/modules/reactor/src/meshdivisions/HexagonalGridDivision.C:158:30: note: read of non-const variable 'INVALID_DIVISION_INDEX' is not allowed in a constant expression
  constexpr auto not_found = MooseMeshDivision::INVALID_DIVISION_INDEX;
```

lindsayad replied:
Can you make `MooseMeshDivision::INVALID_DIVISION_INDEX` a constant expression?

GiudGiud replied:
I think the way forward will likely one day be to have the outside index to be a user settable parameter.
So I d rather not change that for now

lindsayad commented on modules/reactor/test/tests/meshdivisions/tests:
```diff
@@ -0,0 +1,21 @@
+      input = 'hexagonal_division_positions.i'
+      csvdiff = 'hexagonal_division_positions_out.csv'
+      detail = 'hexagonal grids, centered on user-input positions.'
+      # Steady solve
+      recover = false
```
recover generally works with a steady solve

GiudGiud replied:
umm I dont think so? It just errors in a non-error way and passes the test iirc

GiudGiud replied:
like this
```
executioners/executioner.test_steady_adapt: Working Directory: /Users/catherinegiudicelli/projects/moose_v5/test/tests/executioners/executioner
executioners/executioner.test_steady_adapt: Running command: /Users/catherinegiudicelli/projects/moose_v5/test/moose_test-devel -i steady-adapt.i --recover --recoversuffix cpr --error --error-override --no-gdb-backtrace
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: *** Info ***
executioners/executioner.test_steady_adapt: Using out_steady_adapt_cp/0004 for recovery.
executioners/executioner.test_steady_adapt: Framework Information:
executioners/executioner.test_steady_adapt: MOOSE Version:           git commit 93a15fdc00 on 2024-01-06
executioners/executioner.test_steady_adapt: LibMesh Version:         
executioners/executioner.test_steady_adapt: PETSc Version:           3.20.1
executioners/executioner.test_steady_adapt: SLEPc Version:           3.20.0
executioners/executioner.test_steady_adapt: Current Time:            Sat Jan  6 15:49:05 2024
executioners/executioner.test_steady_adapt: Executable Timestamp:    Sat Jan  6 09:22:22 2024
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Parallelism:
executioners/executioner.test_steady_adapt:   Num Processors:          1
executioners/executioner.test_steady_adapt:   Num Threads:             1
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Mesh: 
executioners/executioner.test_steady_adapt:   Parallel Type:           replicated
executioners/executioner.test_steady_adapt:   Mesh Dimension:          2
executioners/executioner.test_steady_adapt:   Spatial Dimension:       2
executioners/executioner.test_steady_adapt:   Nodes:                   157
executioners/executioner.test_steady_adapt:   Elems:                   132
executioners/executioner.test_steady_adapt:   Num Subdomains:          1
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Nonlinear System:
executioners/executioner.test_steady_adapt:   Num DOFs:                157
executioners/executioner.test_steady_adapt:   Num Local DOFs:          157
executioners/executioner.test_steady_adapt:   Num Constrained DOFs:    8
executioners/executioner.test_steady_adapt:   Local Constrained DOFs:  8
executioners/executioner.test_steady_adapt:   Variables:               "u" 
executioners/executioner.test_steady_adapt:   Finite Element Types:    "LAGRANGE" 
executioners/executioner.test_steady_adapt:   Approximation Orders:    "FIRST" 
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Execution Information:
executioners/executioner.test_steady_adapt:   Executioner:             Steady
executioners/executioner.test_steady_adapt:   Solver Mode:             Preconditioned JFNK
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Cannot recover steady solves!
executioners/executioner.test_steady_adapt: Exiting...
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: WARNING! There are options you set that were not used!
executioners/executioner.test_steady_adapt: WARNING! could be spelling mistake, etc!
executioners/executioner.test_steady_adapt: There are 6 unused database options. They are:
executioners/executioner.test_steady_adapt: Option left: name:--error (no value) source: command line
executioners/executioner.test_steady_adapt: Option left: name:--error-override (no value) source: command line
executioners/executioner.test_steady_adapt: Option left: name:--no-gdb-backtrace (no value) source: command line
executioners/executioner.test_steady_adapt: Option left: name:--recover (no value) source: command line
executioners/executioner.test_steady_adapt: Option left: name:--recoversuffix value: cpr source: command line
executioners/executioner.test_steady_adapt: Option left: name:-i value: steady-adapt.i source: command line
executioners/executioner.test_steady_adapt: Running exodiff: /Users/catherinegiudicelli/projects/moose_v5/framework/contrib/exodiff/exodiff -m  -F 1e-10 -t 5.5e-06  /Users/catherinegiudicelli/projects/moose_v5/test/tests/executioners/executioner/gold/out_steady_adapt.e-s004 /Users/catherinegiudicelli/projects/moose_v5/test/tests/executioners/executioner/out_steady_adapt.e-s004
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt:    *****************************************************************
executioners/executioner.test_steady_adapt:               EXODIFF	(Version: 2.90) Modified: 2018-02-15
executioners/executioner.test_steady_adapt:               Authors:  Richard Drake, rrdrake@sandia.gov           
executioners/executioner.test_steady_adapt:                         Greg Sjaardema, gdsjaar@sandia.gov          
executioners/executioner.test_steady_adapt:               Run on    2024/01/06   15:49:05 CET
executioners/executioner.test_steady_adapt:    *****************************************************************
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Reading first file ... 
executioners/executioner.test_steady_adapt: Reading second file ... 
executioners/executioner.test_steady_adapt:   FILE 1: /Users/catherinegiudicelli/projects/moose_v5/test/tests/executioners/executioner/gold/out_steady_adapt.e-s004
executioners/executioner.test_steady_adapt:    Title: out_steady_adapt_0003.e
executioners/executioner.test_steady_adapt:           Dim = 2, Blocks = 1, Nodes = 157, Elements = 132, Nodesets = 4, Sidesets = 4
executioners/executioner.test_steady_adapt:           Vars: Global = 1, Nodal = 1, Element = 0, Nodeset = 0, Sideset = 0, Times = 1
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt:   FILE 2: /Users/catherinegiudicelli/projects/moose_v5/test/tests/executioners/executioner/out_steady_adapt.e-s004
executioners/executioner.test_steady_adapt:    Title: out_steady_adapt.e-s004
executioners/executioner.test_steady_adapt:           Dim = 2, Blocks = 1, Nodes = 157, Elements = 132, Nodesets = 4, Sidesets = 4
executioners/executioner.test_steady_adapt:           Vars: Global = 1, Nodal = 1, Element = 0, Nodeset = 0, Sideset = 0, Times = 1
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: exodiff: INFO .. Map option is not needed.
executioners/executioner.test_steady_adapt: Maximum difference between nodal coordinates = 5.90092e-16
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Nodal coordinates will be compared .. tol:    1e-06 (absolute), floor:        0
executioners/executioner.test_steady_adapt: Time step values will be compared .. tol:  5.5e-06 (relative), floor:    1e-10
executioners/executioner.test_steady_adapt: Global variables to be compared:
executioners/executioner.test_steady_adapt: 	l2_err                           tol:  5.5e-06 (relative), floor:    1e-10
executioners/executioner.test_steady_adapt: Nodal variables to be compared:
executioners/executioner.test_steady_adapt: 	u                                tol:  5.5e-06 (relative), floor:    1e-10
executioners/executioner.test_steady_adapt: No Element variables on either file.
executioners/executioner.test_steady_adapt: No Element Attribute variables on either file.
executioners/executioner.test_steady_adapt: No Nodeset variables on either file.
executioners/executioner.test_steady_adapt: No Sideset variables on either file.
executioners/executioner.test_steady_adapt: Sideset Distribution Factors will be compared .. tol:    1e-06 (relative), floor:        0
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt:   ==============================================================
executioners/executioner.test_steady_adapt:   NOTE: All node and element ids are reported as global ids.
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: Sideset Distribution Factors:
executioners/executioner.test_steady_adapt:   --------- Time step 1, 4.0000000e+00 ~ 4.0000000e+00, rel diff:  0.00000e+00 ---------
executioners/executioner.test_steady_adapt: Global variables:
executioners/executioner.test_steady_adapt: Nodal variables:
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt: exodiff: Files are the same
executioners/executioner.test_steady_adapt: 
executioners/executioner.test_steady_adapt:  
executioners/executioner.test_steady_adapt ...................................................... [recover] OK

```

I dont see the point of running the recover tests on steady execution it just runs the same as steady

lindsayad replied:
If you are skipping `recover` testing because it is conceptually a bit silly, then I am ok with that. But if you were actually getting test harness failures with `recover = true`, then that would be concerning because as you point out there's nothing that really happens in part 2

GiudGiud replied:
the former

GiudGiud committed:
Address Alex's review
- no trivial references
- reference for point arguments
- reword distribution of mesh divisions doco

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

GiudGiud commented on framework/doc/content/source/meshdivisions/CartesianGridDivision.md:
```diff
@@ -8,6 +8,16 @@ Points that lie outside the Cartesian grid may be assigned to the grid outer bin
 parameter.
 
+Using a [Positions](syntax/Positions/index.md) object as the [!param](/MeshDivisions/CartesianGridDivision/center_positions)
+parameter, multiple Cartesian grids can be created around each position computed by that object. The division index
+returned is for a point closest to position $i$ is then:
```
oh this needs a rebase that s in MOOSE already

GiudGiud replied:
umm looks like not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1984118/) on 4d636dd wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26311/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26311/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 28bf8f4f287b56d948c110e55ea724380f959b56`

moosebuild commented:
Job [Modules debug](https://civet.inl.gov/job/1991404/) on 5eeeb7f : invalidated by @GiudGiud

unrelated

moosebuild commented:
Job [Docker moose-dev](https://civet.inl.gov/job/1991398/) on 5eeeb7f : invalidated by @GiudGiud

unrelated

moosebuild commented:
Job [Conda MOOSE (Linux)](https://civet.inl.gov/job/1991396/) on 5eeeb7f : invalidated by @GiudGiud

unrelated

moosebuild commented:
Job [Modules parallel sweep](https://civet.inl.gov/job/1991430/) on 5eeeb7f : invalidated by @GiudGiud

unrelated

moosebuild commented:
Job [Python 3.8](https://civet.inl.gov/job/1991402/) on 5eeeb7f : invalidated by @GiudGiud

unrelated

lindsayad reviewed (approved):
Lol that modules parallel sweep target doesn't look very good, but not this PR's fault

GiudGiud commented:
Thanks for the review.
Yeah I saw. It s mostly THM work. I m pretty busy rn but I ll take a look when things calm down
